src/mqtt/mqtt_test.c

```diff
@@ -1107,7 +1111,15 @@ TEST( MqttTest, MQTT_Connect_LWT )
             /* Timeout. */
             break;
         }
-            
+        else if( strcmp( incomingInfo.pTopicName, TEST_MQTT_LWT_TOPIC ) == 0 )
```

Comment:
Just to verify - there wouldn't be an issue with a previous process loop setting incomingInfo.topicName to that topic, causing this to break out early, correct?

Reply:
No, there should not be an issue as the LWT topic should only get messages when a connection is closed. So, in a normal sub-pub setting, it will not receive any publish from the server.